src/main/java/com/jcabi/beanstalk/maven/plugin/AbstractBeanstalkMojo.java

```diff
@@ -259,6 +331,76 @@ protected boolean isGreen(final Environment env) {
     }
 
     /**
+     * Validate given text in YAML format.
+     *
+     * @param text YAML text
+     */
+    protected void validYaml(final String text) {
+        new Yaml().load(text);
+    }
+
+    /**
+     * Validate the war file.
+     *
+     * @param zip Zip file
+     * @throws org.apache.maven.plugin.MojoFailureException Thrown, if
+     *  validation fails.
+     */
+    protected void validate(final ZipFile zip) throws MojoFailureException {
+        if (zip.getEntry(".ebextension") == null) {
+            throw new MojoFailureException(
+                ".ebextensions directory does not exist in the WAR file"
+            );
+        }
+        final Enumeration<? extends ZipEntry> entries = zip.entries();
+        while (entries.hasMoreElements()) {
+            final ZipEntry entry = entries.nextElement();
+            if (entry.getName().startsWith(".ebextensions/")
+                && !entry.isDirectory()) {
+                try {
+                    this.validYaml(this.readFile(zip, entry));
+                } catch (final YAMLException exception) {
+                    throw new MojoFailureException(
+                        String.format(
+                            "File '%s' in .ebextensions is not YAML valid. %s",
+                            entry.getName(),
+                            exception.getMessage()
+                        ),
+                        exception
+                    );
+                }
+            }
+        }
+    }
+
+    /**
+     * Reads text from a ZIP file.
+     * @param warf ZIP file, which contains entry.
+     * @param entry ZIP entry (compressed file) to read from.
+     * @return Text content of entry.
+     * @throws MojoFailureException thrown when encounter error.
+     */
+    private String readFile(final ZipFile warf, final ZipEntry entry)
+        throws MojoFailureException {
+        InputStreamReader reader = null;
```

Comment:
@tmjee the use of `null` tells us that the design is wrong, see http://www.yegor256.com/2014/05/13/why-null-is-bad.html After taking a closer look at this code I understood what is wrong. Your try/catch block is too big. This is how this code should look:

```java
final Reader reader;
try {
  reader = new InputStreamReader(warf.getInputStream(entry));
} catch () {
  throw ...
}
try {
  return CharStreams.toString(reader);
} finally {
  reader.close();
}
```

This is how it should be. Keep in mind that every time you see `null` - it's a bad design.